### PR TITLE
Incorporate bundle checkout into GET /bundles/<uuid>

### LINKED ISSUE #1203: GET /bundles/<uuid> with `direct url` option should return links to checkout bucket
## Need (user story)
As a user of the DSS API
I want direct urls to files in the checkout bucket and not the DSS storage bucket
because access to the DSS storage bucket is restricted.

### PR DESCRIPTION
**Pull request by ttung**
This behavior is only triggered for presignedurl=true or directurl=true.

Depends on #1362, #1359, #1371 
Fixes #1203, #1204, and #1289

### CODE REVIEW

File: tests/test_bundle.py [hunk 1]
```diff
@@ -30,6 +30,10 @@
 from tests import get_auth_header
 
 
+BUNDLE_GET_RETRY_COUNT = 60
+"""For GET /bundles requests that require a retry, this is the maximum number of attempts we make."""
```
[the remark this answers is not in the document]
Comment (ttung) [reply tests/test_bundle.py]: > I don't see a sleep or use of an HTTP lib in DSSAssertResponse, so the for loop there will just spin quickly through 60 retries?

In practice it works out to be about a request every .5-1s.  If you think there's value in adding a sleep, I can do that in a separate PR.

[the remark this answers is not in the document]
Comment (mweiden) [reply tests/test_bundle.py]: Not sure. I guess my qualm is that it's difficult to tell how this code will behave by looking at it. Ideas? 💡

[the remark this answers is not in the document]
Comment (ttung) [reply tests/test_bundle.py]: Let me take a stab at addressing that in a separate PR.

[the remark this answers is not in the document]
Comment (mweiden) [reply tests/test_bundle.py]: OK, sounds good. 🚢

[the remark this answers is not in the document]
Comment (ttung) [reply tests/test_bundle.py]: #1371

**Pull request merged**